include/kllvm/ast/AST.h

```diff
@@ -519,16 +597,25 @@ class KOREDefinition {
   KOREObjectVariableMapType objectVariables;
   KOREMetaVariableMapType metaVariables;
   KOREModuleMapType moduleNames;
+  KOREObjectCompositeSortDeclarationMapType sortDeclarations;
+  KOREObjectSymbolDeclarationMapType symbolDeclarations;
 
   std::vector<KOREModule *> modules;
-  std::vector<KOREPattern *> attributes;
+  llvm::StringMap<KOREObjectCompositePattern *> attributes;
```

Comment:
I'm interested in what heuristic you use to decide what should have its own type name and what shouldn't. It seems to be related to type length.